window.c

```diff
@@ -120,39 +170,28 @@ void win_init(win_t *win)
 		e->depth = attr.depth;
 	}
 
+
 	XMatchVisualInfo(e->dpy, e->scr, e->depth, TrueColor, &vis);
 	e->vis = vis.visual;
 	e->cmap = XCreateColormap(e->dpy, parent, e->vis, None);
 
-	if (setlocale(LC_CTYPE, "") == NULL || XSupportsLocale() == 0)
-		error(0, 0, "No locale support");
+	win->win_bg = DEFAULT_WIN_BG;
+	win->win_fg = DEFAULT_WIN_FG;
+	win->mrk_fg = DEFAULT_MARK_FG;
 
-	XrmInitialize();
 	res_man = XResourceManagerString(e->dpy);
 	db = res_man != NULL ? XrmGetStringDatabase(res_man) : None;
 
-	f = win_res(db, RES_CLASS ".bar.font", "monospace-8");
-	win_init_font(e, f);
+#if HAVE_LIBXFT
+	win->bar_bg = DEFAULT_WIN_FG;
+	win->bar_fg = DEFAULT_WIN_BG;
+	xft_init(e, win, db);
+#else
+	set_from_hex_string_if_present(db, RES_CLASS ".window.background", &win->win_bg);
+	set_from_hex_string_if_present(db, RES_CLASS ".window.foreground", &win->win_fg);
+	set_from_hex_string_if_present(db, RES_CLASS ".mark.foreground", &win->mrk_fg);
+#endif
```

Comment:
2  things, wasn't the default changed to the bar bg beign the window bg and the bar fg being the window fg, also the code in the else would leave the bar colors unset wouldn't it?

Reply:
Ah. Missed a sport again. Will flip those to make the bar bg/fg match the win fg/bg even if xresources aren't used.

For the else clause, the bar isn't available without Xft/font config because those are needed to draw the text. I actually remove them from the window stuct in such case